account_check_printing_aeroo/models/account_payment.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from num2words import num2words
+from odoo import api, models
+
+
+class AccountPayment(models.Model):
+
+    _inherit = "account.payment"
+
+    @api.multi
+    def do_print_checks(self):
+        if self.journal_id.check_report_id:
+            return self.env['report'].get_action(
+                self, self.journal_id.check_report_id.report_name)
+        return super(AccountPayment, self).do_print_checks()
+
+    def _get_check_amount_in_words(self, amount):
+        """Add support for French amounts in words.
+
+        The module account_check_printing always translates the
+        amount in english in onchange functions.
+        """
+        lang = self.journal_id.check_report_lang
+
+        amount_in_word = num2words(int(amount), lang=lang or 'en_US')
+        cents = int(amount * 100) % 100
```

Comment:
@joaoalf I understand your point, but one dollar is one dollar. One cent is one cent. You can not create a check with more precision than cents. This would not comply with IFRS.